### PR TITLE
Codegen fixes

### PR DESCRIPTION
**Pull request by ssanjay1**

### CODE REVIEW

File: packages/client/src/objectSet/toConjureObjectSet.ts [hunk 1]
```diff
@@ -74,12 +74,17 @@ export function toConjureObjectSet(
         ),
       });
     case "searchAround":
+    case "asBaseObjectTypes":
```
Comment (ssanjay1) [inline packages/client/src/objectSet/toConjureObjectSet.ts:77]: New objectset types were added

File: packages/foundry.streams/src/_components.ts [hunk 2]
```diff
@@ -59,6 +59,29 @@ export interface Dataset {
...
+  records: Array<_Record>;
+  viewRid?: ViewRid;
+}
+
+/**
+ * A record to be published to a stream.
+ *
+ * Log Safety: DO_NOT_LOG
+ */
+export type _Record = Record<string, any>;
```
Comment (ssanjay1) [inline packages/foundry.streams/src/_components.ts:83]: So they declared a type called Record, which is why I had to do some weird overrides in our codegen to not conflict with TS record types

File: packages/platform-sdk-generator/src/addPackagesToPackageJson.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export async function addPackagesToPackageJson(
     packageJson[section],
     Object.fromEntries(packagesToAdd.map(a => [a, "workspace:*"])),
   );
-  packageJson.dependencies["@osdk/shared.client"] = "workspace:~";
+  packageJson.dependencies["@osdk/shared.client"] = "1.0.1";
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/addPackagesToPackageJson.ts:29]: We removed shared.client so just made it concrete dep on 1.0.1 that we released

File: packages/platform-sdk-generator/src/generatePlatformSdkv2.ts [hunk 4]
```diff
@@ -161,7 +161,7 @@ export async function generateComponents(
       continue;
     }
     out += component.getDeclaration(ns.name);
-    ret.push(component.name);
+    ret.push(component.name === "Record" ? "_Record" : component.name);
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/generatePlatformSdkv2.ts:164]: To deconflict Record

File: packages/platform-sdk-generator/src/generateResource2.ts [hunk 5]
```diff
@@ -58,7 +58,9 @@ async function generateMethods(resource: Resource, model: Model) {
   const referencedTypes = new Set<Component>();
 
   for (const method of resource.operations) {
-    const methodName = (method.verb === "delete") ? method.name : method.verb;
+    const methodName = (method.verb === "delete" || method.verb === "load")
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/generateResource2.ts:61]: The new loadObject set endpoints for interfaces also use verb load so they conflict in codegen. Used name instead to deconflict

File: packages/platform-sdk-generator/src/model/Component.ts [hunk 6]
```diff
@@ -67,9 +67,11 @@ export class Component extends Type {
    * 
    * Log Safety: ${component.safety}
    */
-  export ${isAlias ? "type" : "interface"} ${component.locator.localName} ${
-      isAlias ? " = " : ""
-    }`;
+  export ${isAlias ? "type" : "interface"} ${
+      component.locator.localName === "Record"
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/model/Component.ts:71]: More record type deconflicting

File: packages/platform-sdk-generator/src/model/Model.ts [hunk 7]
```diff
@@ -94,6 +94,8 @@ export class Model {
         throw new Error("Method not implemented.");
       case "binary":
         return new BinaryType();
+      case "string":
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/model/Model.ts:97]: Maps can have a key of raw string type, we never ran into this before but the new stream apis added something with a Map<string, othertype> so had to add string type here

File: packages/platform-sdk-generator/src/model/Type.ts [hunk 8]
```diff
@@ -22,7 +22,9 @@ export abstract class Type {
   abstract getTsReferenceString(localNamespace: string | undefined): string;
 
   getPropertyDeclaration(name: string, localNamespace: string | undefined) {
-    return `${name}: ${this.getDeclaration(localNamespace)}`;
+    return `${name}: ${
```
Comment (ericanderson) [inline packages/platform-sdk-generator/src/model/Type.ts:25]: This part of this change makes me a little worried. I am guessing that changing `Component`'s `getTsReferenceString` to check for `Record` might be the more stable change here. I think you can then also replace the change to `Component`'s `getDeclaration` to just call its own `getTsReferenceString` when getting the name.

**Review by ericanderson — CHANGES_REQUESTED** [review]

File: packages/platform-sdk-generator/src/generateResource2.ts (hunk 5 above)
Comment (ssanjay1) [reply packages/platform-sdk-generator/src/generateResource2.ts:61]: Discussing offline

File: packages/platform-sdk-generator/src/model/Type.ts (hunk 8 above)
Comment (ssanjay1) [reply packages/platform-sdk-generator/src/model/Type.ts]: Done

File: packages/platform-sdk-generator/src/model/Component.ts [hunk 9]
```diff
@@ -49,10 +49,14 @@ export class Component extends Type {
   }
 
   getTsReferenceString(localNamespace: string) {
+    const revisedName = this.component.locator.localName.replace(
```
[the remark this answers is not in the document]
Comment (ericanderson) [reply packages/platform-sdk-generator/src/model/Component.ts:52]: Im thinking a `if (this.component.locator.localName === "Record")` probably would have sufficed :)

**Review by ericanderson — APPROVED** [review]

File: packages/platform-sdk-generator/src/model/Component.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (ssanjay1) [reply packages/platform-sdk-generator/src/model/Component.ts:52]: *face palms*

**Review by ericanderson — CHANGES_REQUESTED** [review]
We can close this right

**Comment by ssanjay1** [comment]
Deleted platform APIs

**Pull request closed without merging**